packages/vscode-extension/src/debug/prerequisitesHandler.ts

```diff
@@ -175,6 +174,47 @@ async function runWithCheckResultsTelemetry(
   );
 }
 
+// Mainly addresses two issues:
+// 1. Some error messages contain special characters which will cause the whole debug-check-results to be redacted.
+// 2. CheckResult[] is hard to parse in kusto query (an array of objects).
+//
+// `debug-check-results` contains only known content and we know it will not be redacted.
+// `debug-check-results-raw` might contain arbitrary string and be redacted.
+function convertCheckResultsForTelemetry(checkResults: CheckResult[]): [string, string] {
+  const resultRaw: { [checker: string]: unknown } = {};
+  const resultSafe: { [checker: string]: { [key: string]: string | undefined } } = {};
+  for (const checkResult of checkResults) {
+    resultRaw[checkResult.checker] = checkResult;
+    resultSafe[checkResult.checker] = {
+      result: checkResult.result,
+      source: checkResult.error?.source,
+      errorCode: checkResult.error?.name,
+      errorType:
+        checkResult.error === undefined
+          ? undefined
+          : checkResult.error instanceof UserError
+          ? "user"
+          : checkResult.error instanceof SystemError
+          ? "system"
+          : "unknown",
+    };
+  }
+
+  return [JSON.stringify(resultRaw), JSON.stringify(resultSafe)];
+}
+
+function addCheckResultsForTelemetry(
+  checkResults: CheckResult[],
+  properties: { [key: string]: string },
+  errorProps: string[]
+): void {
+  const [resultRaw, resultSafe] = convertCheckResultsForTelemetry(checkResults);
+  properties[TelemetryProperty.DebugCheckResults] = resultSafe;
+  properties[TelemetryProperty.DebugCheckResultsRaw] = resultRaw;
```

Comment:
`DebugCheckResultsRaw` is previous `DebugCheckResults`? How about keep `DebugCheckResults` as raw, but use something like `debug-check-results-safe`?

Reply:
`debug-check-results` is added last week. This will not affect existing data.